### PR TITLE
Howto doc for manually adding routing tiles

### PR DESCRIPTION
**Pull request by stanw-cg**
Howto doc for manually adding routing tiles and the basic information for routing tiles.

### CODE REVIEW

File: docs/HOWTO_add_routing_tiles.MD [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	Following are the python code for the decode function. More details and full python script can be found from: https://valhalla.readthedocs.io/en/latest/tiles/
+
+	#!/usr/bin/env python
+
+	LEVEL_BITS = 3 
+	TILE_INDEX_BITS = 22 
+	TILE_INDEX_MASK = (2**TILE_INDEX_BITS) - 1
+	def get_tile_index(id): 
+	  return (id >> LEVEL_BITS) & TILE_INDEX_MASK
```
Comment (treee111) [inline docs/HOWTO_add_routing_tiles.MD:49]: same here, codeblock could help!

File: docs/HOWTO_add_routing_tiles.MD [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	    | - ……
+	    | - ……
+	    | - xxx
+	    |    | - yyy.map.lzma
+	    |    | -  ……
+	    |    | - yyy.map.lzma
+
+
+2. How to get routing tiles?
+	a. Option a: download from wahoo update server
```
Comment (treee111) [inline docs/HOWTO_add_routing_tiles.MD:38]: the indentation is not reflected in Github.
Lists could help: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#lists

File: docs/HOWTO_add_routing_tiles.MD [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	def get_tile_index(id): 
+	  return (id >> LEVEL_BITS) & TILE_INDEX_MASK
+
+	After decoding the routing tile IDs into index, it is fairly easy to download all routing tiles from wahoo update server. Following are the URL:
+		https://cdn.wahooligan.com/wahoo-maps/{version}/routing/tiles/{level}/000/{x}/{y}.gph.lzma
+	And the latest map version is 12; level is 2; x, y are both 3 digitals map tile index.
+
+	Note:
+			i. Please use 3 digitals for x and y, and add "0" if x or y is less than 100. e.g. use "003" instead "3" when trying to download them
+			ii. Some tiles may be missing, and the reason and impact are unclear. But it still can work.
```
Comment (treee111) [inline docs/HOWTO_add_routing_tiles.MD:57]: the indentation is not reflected in Github.
Lists or empty lines could help in my opinion

File: docs/HOWTO_add_routing_tiles.MD [hunk 4]
```diff
@@ -0,0 +1,62 @@
+HOWTO_Add_routing tiles for Wahoo Elemnt ROAM
```
Comment (treee111) [inline docs/HOWTO_add_routing_tiles.MD:1]: could we rename the file to HOWTO_ADD_ROUTING_TILES.md to make it identically to the other docs/ files?

**Review by treee111 — CHANGES_REQUESTED** [review]
thanks for the howto-doc @stanw-cg!
Very detailed and understandable :-)

I added some comments to make it better readable directly on Github.
It would be great if you integrate them.
If you don't have time or don't like to, I could also do that!

File: docs/HOWTO_add_routing_tiles.MD (hunk 4 above)
Comment (stanw-cg) [reply docs/HOWTO_add_routing_tiles.MD:1]: Thanks a lot for the great suggestion! Let's update the doc.

File: docs/HOWTO_add_routing_tiles.MD [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	    |    | - yyy.map.lzma
+	    |    | -  ……
+	    |    | - yyy.map.lzma
+	    | - ……
+	    | - ……
+	    | - ……
+	    | - xxx
+	    |    | - yyy.map.lzma
+	    |    | -  ……
+	    |    | - yyy.map.lzma
```
[the remark this answers is not in the document]
Comment (stanw-cg) [reply docs/HOWTO_add_routing_tiles.MD:34]: Sure. Thanks! Let me check and update.

**Comment by stanw-cg** [comment]
Thanks! Let me update the doc.

**Comment by stanw-cg** [comment]
@treee111 Sorry for the multiple commits due to not familiar with markdown syntax :) Please help to review and let me know if further changes are required. Thanks!

**Comment by treee111** [comment]
Hi @stanw-cg, don't worry about the commits! Later on we can stash all your commits into one! 👍 

There is also a Plugin to preview Markdown in VSCode (or maybe also another editor you use): https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one

**Comment by treee111** [comment]
thanks for updating everything, great work @stanw-cg!
It looks fine in my opinion.

If you are also satisfied, I gonna merge that PR?

**Review by treee111 — APPROVED** [review]
thanks for updating everything, great work @stanw-cg!
It looks fine in my opinion.

If you are also satisfied, I gonna merge that PR?

**Comment by stanw-cg** [comment]
@treee111 Thanks a lot!

**Pull request merged**